pkg/client/cache/listwatch.go

```diff
@@ -60,9 +61,7 @@ func NewListWatchFromClient(c Getter, resource string, namespace string, fieldSe
 			Prefix("watch").
 			Namespace(namespace).
 			Resource(resource).
-			// TODO: Use VersionedParams once this is supported for non v1 API.
-			Param("resourceVersion", options.ResourceVersion).
-			TimeoutSeconds(timeoutFromListOptions(options)).
+			VersionedParams(&options, api.Scheme).
```

Comment:
IIUC, this call will end up calling the conversion functions added in pkg/api/conversion.go (https://github.com/kubernetes/kubernetes/pull/17823/files#diff-380d3fffd8583e3e031622a90dcaec36R138), which does not involve a "version", all the conversion function does is a nil check and checking if the option is legit by round-tripping it.

If the above is true, then I think:
1. The function name "VersionedParams" seems misleading as we are using the same unversioned.ListOptions type in all groups.
2. Can we call the conversion function directly instead of going through several indirections in the convertor? because after all, we are converting unversioned.ListOptions to unversioned.ListOptions.

I won't block this PR, just want to discuss if the above suggestions make sense.